### PR TITLE
[Enhancement] Avoid duplicate schema file creation

### PR DESCRIPTION
**Pull request by sduzh**
For each partition, the schema file is created only when the first tablet is created.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -345,7 +345,9 @@ Status TabletManager::create_tablet(const TCreateTabletReq& req) {
     RETURN_IF_ERROR(starrocks::convert_t_schema_to_pb_schema(
             req.tablet_schema, next_unique_id, col_idx_to_unique_id, tablet_metadata_pb->mutable_schema(),
             req.__isset.compression_type ? req.compression_type : TCompressionType::LZ4_FRAME));
-    RETURN_IF_ERROR(create_schema_file(req.tablet_id, tablet_metadata_pb->schema()));
+    if (req.create_schema_file) {
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:348]:
```suggestion
    if (req.__isset.create_schema_file && req.create_schema_file) {
```

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:348]: The thrift field `create_schema_file` is defined with default value `true`

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/CreateReplicaTask.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/RollupJobV2.java|1|1|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**